tests/runtests.pl

```diff
@@ -792,6 +793,12 @@ sub stopserver {
         # given a ssh server, also kill socks piggybacking one
         push @killservers, "socks${2}";
     }
+    if($server eq "http") {
+        # since the http2 server is a proxy that needs to know about the
+        # dynamic http port it too needs to get restarted when the http server
+        # is killed
+        push @killservers, "http/2";
```

Comment:
I'm not sure if that's the best approach. Maybe my suggestion is a bit too much for a such a simple task but I would create an associative array with `http` as key where the value is an array of all versions of that specific protocol. This approach is more and easier extensible.

Reply:
We could of course change this concept and store some sort of generic dependency list per server instead, but that would mean a much larger change and I'm not seeing any obvious benefit with that.

Reply:
You're right, we can switch to that concept once we need to handle more protocol version servers